falcon_kit/mains/ovlp_to_graph.py

```diff
@@ -1146,55 +1193,69 @@ def main(argv=sys.argv):
-                rs = reverse_end(t)
-                rt = reverse_end(s)
-                rv = reverse_end(v)
-                edges_to_remove.add( (rs, rt, rv) )
-                length, score, edges, type_ = u_edge_data[ (rs, rt, rv) ]
-                u_edge_data[ (rs, rt, rv) ] = length, score, edges, "spur:2"
 
-    for n in t_nodes:
-        if ug.out_degree(n) != 0:
-            continue
-        for s, t, v in ug.in_edges(n, keys=True):
-            length, score, edges, type_ = u_edge_data[ (s, t, v) ]
-            if length > 50000 and len(edges) > 3:
+            with_extern_node = False
+            b_in_nodes = [e[0] for e in ug2.in_edges(b_node)]
+            if len(b_in_nodes) == 1:
                 continue
-            in_degree = len( set( e[0] for e in ug.in_edges(s))  ) # ignore mutli-edges
-            out_degree = len( set( e[1] for e in ug.out_edges(s)) )
-            if in_degree > 0 and out_degree > 1:
-                spur_edges.add( (s, t, v) )
-                edges_to_remove.add( (s, t, v) )
-                u_edge_data[ (s, t, v) ] = length, score, edges, "spur:2"
-                rs = reverse_end(t)
-                rt = reverse_end(s)
-                rv = reverse_end(v)
-                edges_to_remove.add( (rs, rt, rv) )
-                length, score, edges, type_ = u_edge_data[ (rs, rt, rv) ]
-                u_edge_data[ (rs, rt, rv) ] = length, score, edges, "spur:2"
+            for v in  b_in_nodes:
+                if v not in n_egg_node_set:
+                    with_extern_node = True
+                else:
+                    continue
+
+            if with_extern_node:
+                s_path = nx.shortest_path( ug2, n, b_node )
+                v1 = s_path[0]
+                total_length = 0
+                for v2 in s_path[1:]:
+                    for s, t, v in ug2.out_edges(v1, keys=True):
+                        if t != v2:
+                           continue
+                        length, score, edges, type_ = u_edge_data[ (s, t, v) ]
+                        total_length += length
+                    v1 = v2
+                if total_length < 50000:
+                    v1 = s_path[0]
+                    for v2 in s_path[1:]:
+                        for s, t, v in ug2.out_edges(v1, keys=True):
+                            if t != v2:
+                               continue
+                            length, score, edges, type_ = u_edge_data[ (s, t, v) ]
+                            rs = reverse_end(t)
+                            rt = reverse_end(s)
+                            rv = reverse_end(v)
+                            try:
+                                ug2.remove_edge( s, t, key= v)
+                                ug2.remove_edge( rs, rt, key= rv)
+                                u_edge_data[ (s, t, v) ] = length, score, edges, "spur:2"
+                                u_edge_data[ (rs, rt, rv) ] = length, score, edges, "spur:2"
+                            except:
```

Comment:
except Exception:
        pass

Otherwise, you can trap `SyntaxError` and `KeyboardInterrupt`.